test/libsolidity/semanticTests/viaYul/detect_add_overflow.yul

```diff
@@ -0,0 +1,17 @@
+// compileViaYul
+contract C {
+    function f(uint a, uint b) public pure returns (uint x) {
+        x = a + b;
+    }
+    function g(uint8 a, uint8 b) public pure returns (uint8 x) {
+        x = a + b;
+    }
+}
+// ----
```

Comment:
One of the failure cases here seem to succeed...

Reply:
And I don't really see why...

Reply:
It's all of them in fact...

Reply:
Ah! Doh :-D! These don't have the ``compileViaYul`` moved down to the expectation blocks, so they're not testing this PR at all :-D.

Reply:
Just moved them down myself.